### PR TITLE
Upgrade Rttov13 + WRF changes

### PR DESCRIPTION
**Pull request by lkugler**
## Description:

- Mostly changes to allow the use of RTTOV13

Edit (@hkershaw-brown) WRF changes moved to https://github.com/NCAR/DART/tree/wrf-thm
- WRF temperature variable changed from T to THM in WRF's model_mod
- Allow interpolate to access the uppermost level of the WRF state.

### Fixes issue

- #366 
- #365 
- RTTOV12->13 upgrade has no issue

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue): #365 
- [x] New feature (non-breaking change which adds functionality): RTTOV upgrade
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected): RTTOV-SCATT is not tested! Code missing! 
- [ ] Documentation update

### Documentation changes needed?

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### Tests

- rttov13 changes: comparison to my own script using the python interface of RTTOV
- WRF allow accessing the top level of the WRF state in interpolate: ./perfect_model_obs gave the expected results
- cange to THM as type_t variable: ./perfect_model_obs did not crash

## Checklist for merging

- [ ] Updated changelog entry
- [ ] Documentation updated
- [ ] Version tag 

## Testing Datasets

- [x] Dataset needed for testing has been provided: 
wrfout state, expected output in netcdf format, obs_seq.in

### CODE REVIEW

**Comment by lkugler** [comment]
### Before-image
Comparison before
### After-image VIS
Comparison after VIS

### After-image WV
Comparison after WV

**Comment by nancycollins** [comment]
by the way, really nice catch on the inverted column(s).

**Review by hkershaw-brown — CHANGES_REQUESTED** [review]
Hi Lukas, 

This is great work to get the rttov13 interface up and running. 

For the  obs_def_rttov13_mod.f90, at the moment git is interpreting this as a `git mv` so 
obs_def_rttov_mod.f90 no longer exists in this branch.

Because we have people using rttov 12, for this pull request I'd like to have

obs_def_rttov_mod.f90
obs_def_rttov13_mod.f90 

I think this makes clearer if we do depreciate rttov12 in a future release.

The mkmf.template can have a VERSION variable so we do not need two mkmf.template.rttov for each compiler:

e.g.
```
RTTOV_VERSION = 13

RTLIBS = -lrttov$(RTTOV_VERSION)_wrapper
```

I can go ahead and make these two changes if you are comfortable with me editing your branch. 

Cheers,
Helen

File: models/wrf/model_mod.f90 [hunk 1]
```diff
@@ -7402,7 +7402,7 @@ subroutine fill_default_state_table(default_table)
                           'UPDATE                    ', &
                           '999                       '  /)
 row = row+1
-default_table(:,row) = (/ 'T                         ', &
+default_table(:,row) = (/ 'THM                       ', &
```
Comment (hkershaw-brown) [inline models/wrf/model_mod.f90:7405]: note, T vs THM is a wrf v3 to v4 issue. T is prognostic in v4.

Inside WRF, use_theta_m=0 for T= THM

**Comment by lkugler** [comment]
We can also remove the commit related to the levels (https://github.com/NCAR/DART/pull/367/commits/25efb6c8ac49e5322300e0a5dd2244dc61e157bb) from this PR?! Since its a separate thing.
Following the discussion in #366

**Comment by hkershaw-brown** [comment]
I'm splitting the WRF and RTTOV parts of this pull request.

The WRF changes are on https://github.com/NCAR/DART/tree/wrf-thm

**Review by hkershaw-brown — APPROVED** [review]
Approved.

Splitting the WRF updates to a separate pull request to support wrf3 and wrf4 users.

**Pull request merged**